src/librustc/hir/map/collector.rs

```diff
@@ -127,15 +128,18 @@ impl<'a, 'hir> NodeCollector<'a, 'hir> {
                                                   codemap: &CodeMap,
                                                   commandline_args_hash: u64)
                                                   -> (Vec<MapEntry<'hir>>, Svh) {
-        let mut node_hashes: Vec<_> = self
+        let mut hir_body_nodes = self
             .hir_body_nodes
-            .iter()
-            .map(|&(def_path_hash, dep_node_index)| {
-                (def_path_hash, self.dep_graph.fingerprint_of(dep_node_index))
-            })
-            .collect();
+            .to_vec();
```

Comment:
`to_vec()` will copy the whole vector. However, since we take `self` by value here, we can safely modify it in place. If you declare `self` mutable, then you should be able to do the following:
```rust
self.hir_body_nodes.sort_unstable_by(|&(ref d1, _), &(ref d2, _)| d1.cmp(d2));
let node_hashes = self.hir_body_nodes
    .iter()
    ...
```

Reply:
thank you for review, I was not sure if it won't brake anything. However, modifying hir_body_nodes itself will make the code a lot faster.